src/test/java/org/javarosa/core/test/Scenario.java

```diff
@@ -119,6 +118,84 @@ private Scenario(FormDef formDef, FormEntryController formEntryController) {
         this.formEntryController = formEntryController;
     }
 
+    // region Miscelaneous
+
+    public static TreeReference getRef(String xpath) {
```

Comment:
This method basically just adds support for the position predicate shortcut, right? It's confusing to see this just in tests. Why not actually support https://github.com/opendatakit/javarosa/issues/422?

Reply:
Huh, I didn't know #422 was in the pipeline. We should totally support that, although it feels a change too big for this PR.

I just wanted to be sure that the refs we work with in tests are exactly the same thing that would be produced by the runtime. As you know, equality rules in tree refs are tricky and I was losing confidence on the previous `xpath -> TreeReference` factories that we were using in tests before this.

Also, after this PR gets merged I could send in another for #422 based on the `getRef` implementation if you feel like we're going in the right direction.

Reply:
> I could send in another for #422 

Let's plan on that.